struct with tags

The following program `sample.go` triggers a panic:

```go
package main
  
import "fmt"
    
type Config struct {
    Users        `json:"users,omitempty" mapstructure:","`
    UsersFile    string `json:"usersFile,omitempty"`
    Realm        string `json:"realm,omitempty"`
    RemoveHeader bool   `json:"removeHeader,omitempty"`
    HeaderField  string `json:"headerField,omitempty" export:"true"`
}

// Users holds a list of users
type Users []string 
    
func CreateConfig() *Config {
    return &Config{}
}
    
func main() {
    c := CreateConfig()
    fmt.Println(c)
}
```

Expected result:
```console
$ go run ./sample.go
&{[]   false }
```

Got:
```console
$ yaegi ./sample.go
panic: reflect.StructOf: duplicate field Xstring

goroutine 1 [running]:
reflect.StructOf(0xc00009aa00, 0x9, 0x11, 0x0, 0x0)
	/usr/local/Cellar/go/1.12.5/libexec/src/reflect/type.go:2523 +0x3362
github.com/containous/yaegi/interp.(*Type).TypeOf(0xc0001d48f0, 0xc0001d48f0, 0xc0001d5040)
	/Users/marc/go/src/github.com/containous/yaegi/interp/type.go:666 +0x715
github.com/containous/yaegi/interp.(*Type).TypeOf(0xc0001d5380, 0x40, 0x30)
	/Users/marc/go/src/github.com/containous/yaegi/interp/type.go:659 +0x857
github.com/containous/yaegi/interp.(*Type).frameType(0xc0001d5380, 0x2640d98, 0x0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/type.go:707 +0x13f
github.com/containous/yaegi/interp.(*Scope).add(0xc0000a87c0, 0xc0001d5380, 0xc0001ea900)
	/Users/marc/go/src/github.com/containous/yaegi/interp/scope.go:154 +0x49
github.com/containous/yaegi/interp.(*Interpreter).Cfg.func1(0xc0001ea200, 0xc0001b1d50)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:231 +0x2250
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001ea200, 0xc0001b1d50, 0xc0001b1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:106 +0xc9
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001e6000, 0xc0001b1d50, 0xc0001b1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Interpreter).Cfg(0xc00016e120, 0xc0001e6000, 0xc0000a6068, 0x4, 0x0, 0x0, 0x4)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:38 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00016e120, 0xc0001e0000, 0x1df, 0x3df, 0xc0001e0000, 0x1df, 0x0, 0xc0001bda00)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:248 +0x135
main.main()
	/Users/marc/go/src/github.com/containous/yaegi/yaegi.go:61 +0x443
```

--- a/_test/struct19.go
+++ b/_test/struct19.go
@@ -1,0 +1,23 @@
+package main
+
+import "fmt"
+
+type Config struct {
+	Users        `json:"users,omitempty" mapstructure:","`
+	UsersFile    string `json:"usersFile,omitempty"`
+	Realm        string `json:"realm,omitempty"`
+	RemoveHeader bool   `json:"removeHeader,omitempty"`
+	HeaderField  string `json:"headerField,omitempty" export:"true"`
+}
+
+// Users holds a list of users
+type Users []string
+
+func CreateConfig() *Config {
+	return &Config{}
+}
+
+func main() {
+	c := CreateConfig()
+	fmt.Println(c)
+}

--- a/interp/type.go
+++ b/interp/type.go
@@ -94,6 +94,7 @@ func (c Cat) String() string {
 // StructField type defines a field in a struct
 type StructField struct {
 	name  string
+	tag   string
 	embed bool
 	typ   *Type
 }
@@ -380,22 +381,36 @@ func nodeType(interp *Interpreter, scope *Scope, n *Node) (*Type, error) {
 		t.cat = StructT
 		var incomplete bool
 		for _, c := range n.child[0].child {
-			if len(c.child) == 1 {
+			switch {
+			case len(c.child) == 1:
 				typ, err := nodeType(interp, scope, c.child[0])
 				if err != nil {
 					return nil, err
 				}
 				t.field = append(t.field, StructField{name: fieldName(c.child[0]), embed: true, typ: typ})
 				incomplete = incomplete || typ.incomplete
-			} else {
+			case len(c.child) == 2 && c.child[1].kind == BasicLit:
+				tag := c.child[1].rval.String()
+				typ, err := nodeType(interp, scope, c.child[0])
+				if err != nil {
+					return nil, err
+				}
+				t.field = append(t.field, StructField{name: fieldName(c.child[0]), embed: true, typ: typ, tag: tag})
+				incomplete = incomplete || typ.incomplete
+			default:
+				var tag string
 				l := len(c.child)
+				if c.lastChild().kind == BasicLit {
+					tag = c.lastChild().rval.String()
+					l--
+				}
 				typ, err := nodeType(interp, scope, c.child[l-1])
 				if err != nil {
 					return nil, err
 				}
 				incomplete = incomplete || typ.incomplete
 				for _, d := range c.child[:l-1] {
-					t.field = append(t.field, StructField{name: d.ident, typ: typ})
+					t.field = append(t.field, StructField{name: d.ident, typ: typ, tag: tag})
 				}
 			}
 		}
@@ -660,7 +675,7 @@ func (t *Type) TypeOf() reflect.Type {
 	case StructT:
 		var fields []reflect.StructField
 		for _, f := range t.field {
-			field := reflect.StructField{Name: exportName(f.name), Type: f.typ.TypeOf()}
+			field := reflect.StructField{Name: exportName(f.name), Type: f.typ.TypeOf(), Tag: reflect.StructTag(f.tag)}
 			fields = append(fields, field)
 		}
 		t.rtype = reflect.StructOf(fields)

